pkgs/development/libraries/science/math/latte-integrale/default.nix

```
@@ -0,0 +1,46 @@
+{ lib
+, stdenv
+, fetchurl
+, fetchpatch
+, gmp
+, ntl
+, cddlib
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "latte-integrale";
+  version = "1.7.6";
+
+  src =
+    let
+      inherit (finalAttrs) version;
+      tag = "version_${lib.replaceStrings ["."] ["_"] version}";
+    in
+    fetchurl {
+      url = "https://github.com/latte-int/latte/releases/download/${tag}/latte-int-${version}.tar.gz";
+      sha256 = "sha256-AGwQ6+XVv9ybFZy6YmSkQyhh/nY84F/oIWJKt9P8IXA=";
+    };
```

```suggestion
stdenv.mkDerivation rec {
  pname = "latte-integrale";
  version = "1.7.6";

  src = fetchurl {
    url = "https://github.com/latte-int/latte/releases/download/version_${lib.replaceStrings ["."] ["_"] version}/latte-int-${version}.tar.gz";
    sha256 = "sha256-AGwQ6+XVv9ybFZy6YmSkQyhh/nY84F/oIWJKt9P8IXA=";
  };
```
finalAttrs does not work with FODs

---

What does not work w.r.t. `finalAttrs` and FODs? Do you have a link to an existing discussion or an example what goes wrong?

---

It will not overwrite src and still use the source corresponding to the old hash like it should which makes finalAttrs here a bit pointless.

And no I don't have a link because I don't know what to search for and how to find it in GitHubs terrible search experience.

---

I think you can overwrite `src` (it fails to compile, but that besides the point):
```nix
{ pkgs ? import ./. { } }:

pkgs.latte-integrale.overrideAttrs (old: {
  version = "1.7.5";
  src = old.src.overrideAttrs (_: {
    outputHash = "sha256-czE58gZRjQog9C0eKAQovpAffSkYmyx4RTgsvVOJU94=";
  });
})
```

https://github.com/NixOS/nixpkgs/pull/119942#issuecomment-1117268449

Either way, I have no strong opinion here; I just find `finalAttrs` conceptually a bit nicer :smiley: